modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -398,46 +398,47 @@ func (suite *KeeperTestSuite) TestChanUpgradeAck() {
 				connectionEnd.State = connectiontypes.UNINITIALIZED
 				path.EndpointA.SetConnection(connectionEnd)
 			},
-			false,
+			connectiontypes.ErrInvalidConnectionState,
 		},
 		{
 			"upgrade not found",
 			func() {
 				store := suite.chainA.GetContext().KVStore(suite.chainA.GetSimApp().GetKey(exported.ModuleName))
 				store.Delete(host.ChannelUpgradeKey(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID))
 			},
-			false,
+			types.ErrUpgradeNotFound,
 		},
 		{
-			"channel end version mismatch on crossing hellos",
+			"startFlushUpgradeHandshake fails due to proof verification failure, counterparty upgrade connection hops are tampered with",
 			func() {
-				channel := path.EndpointA.GetChannel()
-				channel.State = types.TRYUPGRADE
-
-				path.EndpointA.SetChannel(channel)
-
-				upgrade := path.EndpointA.GetChannelUpgrade()
-				upgrade.Fields.Version = "invalid-version"
-
-				path.EndpointA.SetChannelUpgrade(upgrade)
+				counterpartyUpgrade.Fields.ConnectionHops = []string{ibctesting.InvalidID}
 			},
-			false,
+			commitmenttypes.ErrInvalidProof,
 		},
 		{
-			"startFlushUpgradeHandshake fails due to proof verification failure, counterparty upgrade connection hops are tampered with",
+			"startFlushUpgradeHandshake fails due to mismatch in upgrade ordering",
```

Comment:
I changed this testcase from: `"startFlushUpgradeHandshake fails due to mismatch in upgrade sequences"` to this one which fails with ordering mismatch. 

This was the one testcase which was actually failing on proof verification before we got to the checks afterwards. 
We can't force mismatch on upgrade sequences in this test as the counterparty channel upgrade sequence is built using our channel upgrade sequence, so mutating one or the other causes proof verification to fail. Decided to test for mismatch in upgrade ordering instead